authSDK/auth/src/main/kotlin/com/walletconnect/auth/engine/domain/AuthEngine.kt

```
@@ -55,8 +137,109 @@ internal class AuthEngine(
         }
     }
 
-    fun handleInitializationErrors(onError: (WalletConnectException) -> Unit) {
-        relayer.initializationErrorsFlow.onEach { walletConnectException -> onError(walletConnectException) }.launchIn(scope)
+    internal fun respond(
+        respond: EngineDO.Respond,
+        onFailure: (Throwable) -> Unit,
+    ) {
+        if (authRequestMap[respond.id] == null) {
+            Logger.error(MissingAuthRequestException.message)
+            onFailure(MissingAuthRequestException)
+            return
```

You can just throw MissingAuthRequestException instead. It's going to be handle in AuthProtocol and we have less code here.

---

Right